src/Module/Upgrade/Upgrade1_5_0.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gett\MyparcelBE\Module\Upgrade;
+
+use Gett\MyparcelBE\Constant;
+use Gett\MyparcelBE\Database\Table;
+use MyParcelNL\Sdk\src\Support\Collection;
+
+class Upgrade1_5_0 extends AbstractUpgrade
+{
+    private $carrierConfigurationTable;
+
+    /**
+     * @throws \PrestaShopDatabaseException
+     */
+    public function upgrade(): void
+    {
+        $this->addInsuranceOptionsForCarriers();
+    }
+
+    /**
+     * Adds insurance_from_price and insurance_max_amount to carrier configuration if not already present
+     *
+     * @throws \PrestaShopDatabaseException
+     */
+    private function addInsuranceOptionsForCarriers(): void
+    {
+        $this->carrierConfigurationTable = Table::withPrefix(Table::TABLE_CARRIER_CONFIGURATION);
+        $configName                      = Constant::INSURANCE_CONFIGURATION_NAME;
+
+        $query   = <<<SQL
+SELECT DISTINCT id_carrier, name, value FROM $this->carrierConfigurationTable WHERE name LIKE '%$configName'
+SQL;
+        $records = new Collection($this->db->executeS($query));
+
+        foreach ($records as $record) {
+            $carrierId = (int) $record['id_carrier'];
+            $maxAmount = '1' === $record['value']
+                ? 500
+                : 0;
+
+            $optionName = Constant::INSURANCE_CONFIGURATION_FROM_PRICE;
+            if (!$this->optionExists($carrierId, $optionName)) {
```

Comment:
Gezien je addOption toch alleen zo gebruikt, kan je misschien net zo goed meteen de `this->optionExists` check in die functie zelf stoppen

Reply:
Dit is m.i. netter een scheiding van functionaliteit, dan zou je daaroverheen wel een method this->addOptionWhenNotExists kunnen maken, is uiteindelijk beter, maar voor zo'n simpele migratie is dat misschien niet een beetje overkill en de tekst zoals die nu staat niet helderder?